ape_etherscan/client.py

```diff
@@ -82,12 +79,66 @@ def __init__(self, network_name: str, address: str):
     def get_source_code(self) -> Optional[Dict]:
         params = {**self.base_params, "action": "getsourcecode", "address": self._address}
         result = self._get(params=params)
+
+        if not result:
+            raise get_request_error(response)
+
         return result[0] if len(result) == 1 else None
 
 
+class AccountClient(_APIClient):
+    def __init__(self, network_name: str, address: str):
+        self._address = address
+        super().__init__(network_name, "account")
+
+    def get_all_normal_transactions(
+        self,
+        start_block: Optional[int] = None,
+        end_block: Optional[int] = None,
+        offset: int = 100,
+        sort: str = "asc",
+    ) -> Iterator[List[Dict]]:
+        page_num = 1
+        last_page_results = offset  # Start at offset to trigger iteration
+        while last_page_results == offset:
+            page = self._get_page_of_normal_transactions(
+                page_num, start_block, end_block, offset=offset, sort=sort
+            )
+
+            if len(page):
+                yield page
```

Comment:
Feel like it should be an iterator of items in the page, not an iterator of pages. This would just be moving a for loop from the ExplorerAPI impl to here, but would probably read better.

Would also be nice if it were typed items, but not required

Reply:
It is! Just not here. In `explorer.py`, we loop through the pages and yield the `ReceiptAPI`, as per the API, along with doing some other stuff.

https://github.com/ApeWorX/ape-etherscan/pull/13/files#r787347562

Reply:
I like the class serialization + possibly refactoring so that this method yields individual objects.
Though, this whole module is just an implementation detail to the `Explorer` class

Reply:
- [x] Yield individual items from client
- [ ] Return classes from client